src/index.ts

```diff
@@ -1,129 +1,102 @@
-
-  async run() {
-    const { args, flags } = this.parse(DXScannerCommand);
-    debug('cli args')(args);
-    debug('cli flags')(flags);
-    const scanPath = args.path;
-
-    let authorization = flags.authorization ? flags.authorization : this.loadAuthTokenFromEnvs();
-    const json = flags.json;
-    const fail = <PracticeImpact | 'all'>flags.fail;
-
-    const notifier = updateNotifier({ pkg: this.config.pjson });
-    const hrstart = process.hrtime();
-
-    cli.action.start(`Scanning URI: ${scanPath}`);
-
-    const container = createRootContainer({
-      uri: scanPath,
-      auth: authorization,
-      json,
-      fail,
-      recursive: flags.recursive,
-      ci: flags.ci,
-      fix: flags.fix,
-      fixPattern: flags.fixPattern,
-    });
-    const scanner = container.get(Scanner);
-
-    if (flags.init) {
-      await scanner.init(scanPath);
-      process.exit(0);
-    }
-
-    let scanResult = await scanner.scan();
-
-    if (scanResult.needsAuth && !flags.ci) {
-      if (ScanningStrategyDetectorUtils.isGitHubPath(scanPath) || scanResult.serviceType === ServiceType.github) {
-        authorization = await cli.prompt('Insert your GitHub personal access token. https://github.com/settings/tokens\n', {
-          type: 'hide',
-        });
-      } else if (ScanningStrategyDetectorUtils.isBitbucketPath(scanPath) || scanResult.serviceType === ServiceType.bitbucket) {
-        authorization = await cli.prompt(
-          'Insert your Bitbucket credentials (in format "appPassword" or "username:appPasword"). https://confluence.atlassian.com/bitbucket/app-passwords-828781300.html\n',
-          { type: 'hide' },
-        );
-      }
-
-      const container = createRootContainer({
-        uri: scanPath,
-        auth: authorization,
-        json,
-        fail,
-        recursive: flags.recursive,
-        ci: flags.ci,
-        fix: flags.fix,
-        fixPattern: flags.fixPattern,
+import Init from './commands/init';
+import Practices from './commands/practices';
+import _ from 'lodash';
+import updateNotifier from 'update-notifier';
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+const pjson = require('../package.json');
+
+class DXScannerCommand {
+  static async run() {
+    const cmder = new commander.Command();
+
+    // default cmd config
+    cmder.version(pjson.version).name('dx-scanner');
+
+    // cmd: run
+    cmder
+      .command('run [path]')
+      .description('Scan your project for possible DX recommendations')
+      .option(
+        '-a --authorization <authorization>',
+        'credentials to the repository (in format "token" or "username:token"; can be set as ENV variable DX_GIT_SERVICE_TOKEN)',
+        process.env.DX_GIT_SERVICE_TOKEN || process.env.GITHUB_TOKEN,
+      )
+      .option('--ci', 'CI mode', process.env.CI === 'true')
+      .option('-d --details', 'print details in reports')
```

Comment:
Does it work without a comma between -d and --details ?

Reply:
yes, as others commands :)